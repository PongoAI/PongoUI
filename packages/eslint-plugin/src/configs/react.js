/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
