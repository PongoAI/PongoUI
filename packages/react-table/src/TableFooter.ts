export * from './components/TableFooter/index';
