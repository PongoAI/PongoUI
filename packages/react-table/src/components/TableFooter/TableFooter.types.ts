import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type TableFooterSlots = {
  /**
   * The root of the TableFooter.
   */
  root: Slot<'tfoot'>;
};

type TableFooterCommons = {};

export type TableFooterProps = ComponentProps<TableFooterSlots> & TableFooterCommons;

export type TableFooterState = ComponentState<TableFooterSlots> & TableFooterCommons;
