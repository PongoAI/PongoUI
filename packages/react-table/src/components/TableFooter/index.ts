export * from './TableFooter';
export * from './useTableFooterStyles';
export * from './TableFooter.types';
export * from './useTableFooter';
