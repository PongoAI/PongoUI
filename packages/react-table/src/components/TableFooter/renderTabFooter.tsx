import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { TableFooterState, TableFooterSlots } from './TableFooter.types';

export const renderTableFooter = (state: TableFooterState) => {
  const { slots, slotProps } = getSlots<TableFooterSlots>(state);

  return <slots.root {...slotProps.root} />;
};
