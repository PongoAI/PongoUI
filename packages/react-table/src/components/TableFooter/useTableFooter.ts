import * as React from 'react';
import { getNativeElementProps } from '@fluentui/react-utilities';
import type { TableFooterProps, TableFooterState } from './TableFooter.types';

export const useTableFooter = (props: TableFooterProps, ref: React.Ref<HTMLElement>): TableFooterState => {
  const state: TableFooterState = {
    components: {
      root: 'tfoot',
    },
    root: getNativeElementProps('tfoot', {
      ref,
      ...props,
    }),
  };

  return state;
};
