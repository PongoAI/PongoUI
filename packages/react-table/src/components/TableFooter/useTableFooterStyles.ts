import { makeStyles, mergeClasses } from '@griffel/react';
import type { TableFooterState } from './TableFooter.types';

const useRootStyles = makeStyles({
  root: {
    display: 'table-footer-group',
    verticalAlign: 'middle',
  },
});

export const useTableFooterStyles = (state: TableFooterState) => {
  const rootStyles = useRootStyles();

  state.root.className = mergeClasses(rootStyles.root, state.root.className);

  return state;
};
