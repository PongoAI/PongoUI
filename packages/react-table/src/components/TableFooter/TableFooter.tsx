import * as React from 'react';
import { useTableFooter } from './useTableFooter';
import { useTableFooterStyles } from './useTableFooterStyles';
import { renderTableFooter } from './renderTabFooter';
import type { TableFooterProps } from './TableFooter.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The TableFooter component is used to display a footer of data.
 */
export const TableFooter: ForwardRefComponent<TableFooterProps> = React.forwardRef((props, ref) => {
  const state = useTableFooter(props, ref);

  useTableFooterStyles(state);

  return renderTableFooter(state);
});
TableFooter.displayName = 'TableFooter';
