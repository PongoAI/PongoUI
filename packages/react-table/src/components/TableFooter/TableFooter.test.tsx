import * as React from 'react';
import { TableFooter } from './TableFooter';
import { render } from '@testing-library/react';
import { isConformant } from '../../common/isConformant';

describe('TableFooter', () => {
  isConformant({
    Component: TableFooter,
    displayName: 'TableFooter',
  });

  it('renders a basic Table', () => {
    const renderedComponent = render(<TableFooter />);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
