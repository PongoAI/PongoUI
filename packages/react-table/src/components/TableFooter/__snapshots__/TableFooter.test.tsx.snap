// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableFooter renders a basic Table 1`] = `
<body>
  <div>
    <tfoot
      class="___1hw3nzh f15gr2l fmrv4ls"
    />
  </div>
</body>
`;
